msal/application.py

```diff
@@ -143,12 +151,21 @@ def __init__(
                 validate_authority, verify=verify, proxies=proxies, timeout=timeout)
             # Here the self.authority is not the same type as authority in input
         self.token_cache = token_cache or TokenCache()
-        self.client = self._build_client(client_credential, self.authority)
+        self.client = self._build_client(client_credential, self.authority, app_name, app_version)
         self.authority_groups = None
 
-    def _build_client(self, client_credential, authority):
+    def _build_client(self, client_credential, authority, app_name=None, app_version=None):
```

Comment:
There are 2 occurrences of calling `_build_client(...)` and we need to also change another one. (And then we might not choose to change these parameters at all.)